test/test_reductions.py

```diff
@@ -55,19 +56,191 @@ def _rand_shape(dim, min_size, max_size):
+        args, kwargs = next(op.generate_args_kwargs(t, dim=[0, 2]))
+        self.assertEqual(op(t, *args, dim=[0, 2], **kwargs).shape, [3])
+
+    @ops(filter(lambda op: op.supports_multiple_dims, reduction_op_db), allowed_dtypes=[torch.float])
+    def test_dim_multi_keepdim(self, device, dtype, op: ReductionOpInfo):
+        """Tests that dim=[i, j, ...], keepdim=True reduces dimensions i, j, ... to size 1."""
+        t = make_tensor((2, 3, 2), device, dtype)
+        args, kwargs = next(op.generate_args_kwargs(t, dim=[0, 2]))
+        self.assertEqual(op(t, *args, dim=[0, 2], keepdim=True, **kwargs).shape, [1, 3, 1])
+
+    @ops(filter(lambda op: not op.supports_multiple_dims, reduction_op_db), allowed_dtypes=[torch.float])
+    def test_dim_multi_unsupported(self, device, dtype, op: ReductionOpInfo):
+        """Tests that ops claiming to not support multi dim actually don't."""
+        t = make_tensor((2, 3, 2), device, dtype)
+        with self.assertRaises(TypeError):
+            args, kwargs = next(op.generate_args_kwargs(t, dim=[0, 2]))
+            self.assertEqual(op(t, *args, dim=[0, 2], **kwargs).shape, [3])
+
+    @ops(filter(lambda op: op.supports_multiple_dims, reduction_op_db), allowed_dtypes=[torch.float])
+    def test_dim_empty(self, device, dtype, op: ReductionOpInfo):
+        """Tests that dim=[] is a noop"""
+        t = make_tensor((2, 3), device, dtype)
+        args, kwargs = next(op.generate_args_kwargs(t, dim=[]))
+        self.assertEqual(op(t, *args, dim=[], **kwargs), t)
+
+    @ops(filter(lambda op: op.supports_multiple_dims, reduction_op_db), allowed_dtypes=[torch.float])
+    def test_dim_empty_keepdim(self, device, dtype, op: ReductionOpInfo):
+        """Tests that dim=[], keepdim=True is a noop"""
+        t = make_tensor((2, 3), device, dtype)
+        args, kwargs = next(op.generate_args_kwargs(t, dim=[]))
+        self.assertEqual(op(t, *args, dim=[], keepdim=True, **kwargs), t)
+
+    @ops(reduction_op_db, allowed_dtypes=[torch.float])
+    def test_dim_offbounds(self, device, dtype, op: ReductionOpInfo):
+        """Tests that passing an off-bounds dim throws"""
+        t = make_tensor((2, 3), device, dtype)
+        args, kwargs = next(op.generate_args_kwargs(t, dim=2))
+        with self.assertRaises(IndexError):
+            op(t, *args, dim=2, **kwargs)
+
+    @ops(reduction_op_db, allowed_dtypes=[torch.float])
+    def test_dim_offbounds_keepdim(self, device, dtype, op: ReductionOpInfo):
+        """Tests that passing an off-bounds dim and keepdim=True throws"""
+        t = make_tensor((2, 3), device, dtype)
+        args, kwargs = next(op.generate_args_kwargs(t, dim=2))
+        with self.assertRaises(IndexError):
+            op(t, *args, dim=2, keepdim=True, **kwargs)
+
+    @ops(reduction_op_db, allowed_dtypes=[torch.float])
+    def test_empty_slice(self, device, dtype, op: ReductionOpInfo):
+        """Tests for consistent behavior when reducing over an empty slice."""
```

Comment:
Do we want a pointer to a doc or dev wiki or a comment here explaining what the empty slice reduction behavior should be? Or maybe just an elaborated comment here?

Reply:
I'll add more comments and a link to the GitHub issue discussing this